src/control/ros/RosJointStateClient.cpp

```diff
@@ -0,0 +1,106 @@
+#include <aikido/control/ros/RosJointStateClient.hpp>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+//=============================================================================
+RosJointStateClient::RosJointStateClient(
+      dart::dynamics::SkeletonPtr _skeleton,
+      ::ros::NodeHandle _nodeHandle,
+      const std::string& _topicName,
+      size_t capacity)
+  : mSkeleton{std::move(_skeleton)}
+  , mBuffer{}
+  , mCapacity{capacity}
+  , mCallbackQueue{} // Must be before mNodeHandle for order of destruction.
```

Comment:
Nit: I believe this comment should say **after**, which is both correct _and_ what is currently implemented. 😉